### PR TITLE
CMS v2 quality improvements

### PR DESCRIPTION
**Pull request by lkostrowski**
## Scope of the PR

Set of improvements around quality

## Checklist

- [ ] `.github/dependabot.yaml` is up-to date.
- [ ] I added changesets and [read good practices](/.changeset/README.md).

### CODE REVIEW

File: apps/cms-v2/src/modules/providers/builder.io/builder-io-config-form.tsx [hunk 1]
```diff
@@ -146,8 +146,9 @@ const AddFormVariant = () => {
   const { notifySuccess } = useDashboardNotification();
   const { mutate: addProvider } = trpcClient.providersConfigs.addOne.useMutation({
     onSuccess() {
-      notifySuccess("Success", "Updated configuration");
-      push("/configuration");
+      push("/configuration").then(() => {
+        notifySuccess("Success", "Saved configuration");
+      });
```
Comment (peelar) [inline apps/cms-v2/src/modules/providers/builder.io/builder-io-config-form.tsx:151]: **question**: Not the other way around?

1. Show toast
2. Then push

File: apps/cms-v2/src/modules/providers/contentful/contentful-config-form.tsx [hunk 2]
```diff
@@ -29,29 +28,57 @@ const PureForm = ({
...
-    handleSubmit,
-    formState: { errors },
-  } = useForm<FormSchema>({
-    defaultValues: defaultValues,
-    resolver: zodResolver(ContentfulProviderConfig.Schema.Input.omit({ type: true })),
-  });
+  const { notifyError } = useDashboardNotification();
+
+  const { control, getValues, setValue, watch, handleSubmit, setError, clearErrors } =
+    useForm<FormSchema>({
```
Comment (peelar) [inline apps/cms-v2/src/modules/providers/contentful/contentful-config-form.tsx:34]: **suggestion (non-blocking)**: I think you can skip the type casting if you are providing both `defaultValues` and `resolver`.

File: apps/cms-v2/src/modules/providers/contentful/contentful.router.ts [hunk 3]
```diff
@@ -38,7 +38,11 @@ export const contentfulRouter = router({
         space: input.contentfulSpace,
       });
 
-      return client.getEnvironments();
+      return client.getEnvironments().catch((e) => {
+        console.error(e);
```
Comment (peelar) [inline apps/cms-v2/src/modules/providers/contentful/contentful.router.ts:42]: **question**: was this committed intentionally?

**Review by peelar — APPROVED** [review]

File: apps/cms-v2/src/modules/providers/contentful/contentful.router.ts (hunk 3 above)
Comment (lkostrowski) [reply apps/cms-v2/src/modules/providers/contentful/contentful.router.ts:42]: I will replace with logger

**Review by peelar — APPROVED** [review]

**Pull request merged**